### PR TITLE
Ports: Add menu support for Hλlf-Life port

### PR DESCRIPTION
**Pull request by ccapitalK**
This adds partial menu support. Mouse input still doesn't work, but most other components I've tested work, including saving/loading games. There is also a fix for the freetype port.

### CODE REVIEW

File: Ports/halflife/package.sh [hunk 1]
```diff
@@ -2,35 +2,46 @@
...
     # Build the game and client
-    cd ../hlsdk-xash3d-master
+    cd ./hlsdk-xash3d-master
     ./waf build
     cd ../
 
     # Build the engine
     cd ./xash3d-fwgs-master
-   ./waf build
+    ./waf build
```
Comment (sin-ack) [inline Ports/halflife/package.sh:44]: Spacing typo?

File: Ports/halflife/package.sh [hunk 2]
```diff
@@ -59,7 +77,10 @@ post_install() {
 export LD_LIBRARY_PATH=/home/anon/Games/halflife/
 scriptdir=$(dirname "$0")
 cd $scriptdir
-./xash3d
+./xash3d -console
 EOF
     chmod +x ${SERENITY_INSTALL_ROOT}/home/anon/Games/halflife/hl.sh
+
+    echo "Remember to copy the 'valve/' from your half life installation"
```
Comment (sin-ack) [inline Ports/halflife/package.sh:84]: "Half-Life installation" is probably better capitalization here.

File: Ports/halflife/patches/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+## `fix_mainui_wscript.patch`
+
+mainui_cpp-master/wscript: Link against libdl, libxml2
+When building support for menus in vgui, waflib links against
+fontconfig and freetype2. Unfortunately for us, it does so by
+uses the host system's pkgconfig to determine what to link
+against, which stops it from linking against libdl and libxml2
+(which are needed as transitive dependencies for fontconfig and
+freetype2 on serenity). This patch hacks in a check and build
```
Comment (ADKaster) [inline Ports/halflife/patches/ReadMe.md:11]: this should be avoidable by setting `$PKG_CONFIG_PATH` to point into the sysroot

File: Ports/halflife/package.sh [hunk 4]
```diff
@@ -2,35 +2,46 @@
...
 version="1.0.0"
 useconfigure="true"
-depends=("SDL2")
-workdir="xash3d-fwgs-master"
+depends=("SDL2" "fontconfig" "freetype")
+workdir="."
 files="https://github.com/SerenityPorts/xash3d-fwgs/archive/master.tar.gz xash3d_engine.tar.gz
-https://github.com/SerenityPorts/hlsdk-xash3d/archive/master.tar.gz xash3d_hldll.tar.gz"
+https://github.com/SerenityPorts/hlsdk-xash3d/archive/master.tar.gz xash3d_hldll.tar.gz
+https://github.com/ccapitalK/vgui-dev/archive/master.tar.gz vgui-dev.tar.gz
```
Comment (ADKaster) [inline Ports/halflife/package.sh:9]: any reason not to use the repo you forked from? https://github.com/FWGS/vgui-dev Looks like your master is up to date w/theirs.

Comment (ccapitalK) [reply Ports/halflife/package.sh:9]: Good point, it originally looked like I might need to make some changes to get it to build (and that might be true later as I try to get mouse support working), right now there isn't any reason to use it.

File: Ports/halflife/package.sh (hunk 1 above)
Comment (ccapitalK) [reply Ports/halflife/package.sh:44]: Good catch

File: Ports/halflife/package.sh (hunk 2 above)
Comment (ccapitalK) [reply Ports/halflife/package.sh:84]: Yeah, will make this message nicer.

File: Ports/halflife/patches/ReadMe.md (hunk 3 above)
Comment (ccapitalK) [reply Ports/halflife/patches/ReadMe.md]: Oh wow, I was not aware we had a working pkgconfig system. Yeah, this works after fixing the fontconfig port a bit.

**Comment by ccapitalK** [comment]
So the fontconfig port had to be patched, the pkgconfig file it was installing was malformed for the build on serenity (should require libxml2 and libdl transitively). I think there might be some other way to fix the pkgconfig file (that is a bit less hackish), but this seems to be the most sensible way of fixing it I can find.

**Review by Quaker762 — APPROVED** [review]
Looks good! LGTM!

**Comment by bgianfo** [comment]
Sweet! 🕹️

**Pull request merged**